### DISCUSSION TITLE
scheduler question

### DISCUSSION THREAD

**Discussion by rdyvptnm** [aaguiar10/gflows#6] [Q&A] [answered]
# download data then schedule when to redownload
sched = BackgroundScheduler(daemon=True)
sched.add_job(sensor)
sched.add_job(sensor, CronTrigger.from_crontab("0,30 13-20 * * 0-4", timezone=UTC))
sched.start()

This looks like it  schedules the job to run twice every hour, at 0 and 30 minutes past the hour, only on days Sunday through Thursday, and only between 1 PM and 8 PM in the UTC timezone. 

But your website says every 15 minutes.  is your localy copy changed to run every 15 minutes or this is a bug ?

**Comment by aaguiar10** [comment, marked as answer]
The app updates every 30 minutes during the hours specified in the README / website (9am-4:30pm ET), but the data itself is 15 minutes delayed. 

For the scheduler I entered time constraints that follow the [APScheduler documentation ](https://apscheduler.readthedocs.io/en/3.x/modules/triggers/cron.html). Normally 0-4 for crontab days is Sun-Thu but for this module it's Mon-Fri. As for the UTC timezone I've recently removed it to keep both the scheduler and options data consistent with one timezone "America/New_York" and resolve potential issues with daylight saving time. However, you are free to change it to the zone you'd like.

Some examples below:
```
# scheduler is triggered at time specified in timezone
sched.add_job(
    sensor,
    CronTrigger.from_crontab(
        "0,30 9-16 * * 0-4", timezone=timezone("America/New_York")
    ),
)

...

# datetime object is localized with specified zone
todayDate = DateDataParser(settings={"TIMEZONE": "America/New_York"}).get_date_data(
    todayDate
)

...

# expiration time assumed to be 16:00 (4pm ET)
df["ExpirationDate"] = pd.to_datetime(
    df["ExpirationDate"], format="%a %b %d %Y"
).dt.tz_localize(todayDate.date_obj.tzinfo) + timedelta(hours=16)
```

**Reply by rdyvptnm** [reply]
Do you mind sharing the details for complete URL that i can use in my local copy version either its paid or free?
 Its pain to download the csv data manually from the cboe website every time.
